plugins/initLaunchDarkly.ts

```diff
@@ -1,92 +1,24 @@
-import { initialize, LDClient, LDFlagSet, LDOptions, LDUser } from 'launchdarkly-js-client-sdk'
-
+/* eslint-disable valid-jsdoc */
+import { initLdClient } from '../util/feature-flags'
+import { SessionStorageKeys } from '../util/constants'
 // get rid of "element implicitly has an 'any' type..."
 declare const window: any
-
-/**
- * Default flag values when LD is not available.
- * NB: Uses "business-edit" project (per LD client id in config).
- */
-const defaultFlagSet: LDFlagSet = {
-  'banner-text': '', // by default, there is no banner text
-  'enable-digital-credentials': false,
-  'sentry-enable': false, // by default, no sentry logs
-  'show-alert-phone-numbers-firm': false,
-  'supported-business-summary-entities': [],
-  'supported-correction-entities': [],
-  'supported-dissolution-entities': [],
-  'supported-consent-continuation-out-entities': []
-}
-
 /**
- * The Launch Darkly client instance.
+ * This plugin function is called before instantiating the root Vue.js application.
+ * It initializes LaunchDarkly.
  */
-let ldClient: LDClient
+export default async function (context: { $config: { ldClientId: string } }) {
+  // save id to window object for init function
+  window['ldClientId'] = context.$config.ldClientId
 
-/**
- * An async method that initializes the Launch Darkly client.
- */
-async function InitLdClient (): Promise<void> {
-  const envKey: string = window['ldClientId']
+  if (window['ldClientId']) {
+    console.info('Initializing LaunchDarkly...')
 
-  if (envKey) {
-    const user: LDUser = {
-      // since we have no user data yet, use a shared key temporarily
-      key: 'anonymous'
-    }
-    const options: LDOptions = {
-      // fetch flags using REPORT request (to see user data as JSON)
-      useReport: true,
-      // opt out of sending diagnostics data
-      diagnosticOptOut: true,
-      // open streaming connection for live flag updates
-      streaming: true
-    }
+    // initialize LD using local library
+    const ldClient = await initLdClient()
 
-    ldClient = initialize(envKey, user, options)
-
-    try {
-      await ldClient.waitForInitialization()
-    } catch {
-      // shut down client -- `variation()` will return undefined values
-      await ldClient.close()
-      // NB: LD logs its own errors
-    }
+    // also save flags in session for common components (eg, SbcHeader)
+    const allFlags = JSON.stringify(ldClient.allFlags())
+    sessionStorage.setItem(SessionStorageKeys.LaunchDarklyFlags, allFlags)
```

Comment:
1. I'm not certain, but `ldClient` could maybe be falsey if it fails to initialize (eg, no key, network issue, etc)

2. This code stores the flags at time of init, but what if the flags are changed 5 minutes later? The original code (GetFeatureFlag()) fetches the cached FF, and LD keeps its cache up to date using a network stream, ie, real-time flags. Why not use GetFeatureFlag?

Reply:
Also, somewhere (you may be doing this but I don't remember it), you should call `UpdateLdUser()` to register the current user in LD (otherwise it's "anonymous"). This will let you target flags by userid or maybe roles (ie, staff).